File: Python/0984_Last_Value_Map.py
class LastValueMap:
    def __init__(self):
        self.stack = []
        self.lasts = {}

    def set(self, key, value):
        self.lasts[key] = value
        self.stack.append(key)

    def remove(self, key):
        if key in self.lasts:
            del self.lasts[key]

    def getLast(self):
        while self.stack[-1] not in self.lasts:
            self.stack.pop()
        return self.lasts[self.stack[-1]]

File: CPlusPlus/0984_Last_Value_Map.cpp
class LastValueMap {
    public:
    stack<int> stk;
    unordered_map<int, int> lasts;
    LastValueMap() {

    }

    void set(int key, int value) {
        stk.push(key);
        lasts[key] = value;
    }

    void remove(int key) {
        lasts.erase(key);
    }

    int getLast() {
        while (lasts.find(stk.top()) == lasts.end()) {
            stk.pop();
        }
        return lasts[stk.top()];
    }
};
